packages/atomic/src/components/result-lists/result-list-common.tsx

```diff
@@ -0,0 +1,316 @@
+
+  constructor(opts: ResultListCommonOptions) {
+    this.bindings = opts.bindings;
+    this.updateBreakpoints = once((host: HTMLElement) => {
+      updateBreakpoints(host);
+    });
+
+    if (opts.fieldsToInclude) {
+      this.resultListControllerProps = {
+        options: {
+          fieldsToInclude: this.determineAllFieldsToInclude(
+            opts.fieldsToInclude
+          ),
+        },
+      };
+    }
+
+    this.registerResultTemplates(
+      opts.templateElements,
+      opts.includeDefaultTemplate,
+      opts.onReady,
+      opts.onError
+    );
+  }
+
+  set renderingFunction(render: ResultRenderingFunction) {
+    this.render = render;
+  }
+
+  private determineAllFieldsToInclude(
+    configuredFieldsToInclude: string
+  ): string[] {
+    if (configuredFieldsToInclude.trim() === '')
+      return [...EcommerceDefaultFieldsToInclude];
+    return EcommerceDefaultFieldsToInclude.concat(
+      configuredFieldsToInclude.split(',').map((field) => field.trim())
+    );
+  }
+
+  public getTemplate(result: Result) {
+    return this.resultTemplatesManager.selectTemplate(result);
+  }
+
+  public getContentOfResultTemplate(
+    resultOrFolded: Result | FoldedResult
+  ): HTMLElement | DocumentFragment {
+    const result = (resultOrFolded as FoldedResult).result || resultOrFolded;
+    return this.render ? this.render(result) : this.getTemplate(result)!;
+  }
+
+  private makeDefaultTemplate(): ResultTemplate<DocumentFragment> {
+    const content = document.createDocumentFragment();
+    const linkEl = document.createElement('atomic-result-link');
+    content.appendChild(linkEl);
+    return {
+      content,
+      conditions: [],
+    };
+  }
+
+  public async registerResultTemplates(
+    elements: NodeListOf<TemplateElement>,
+    includeDefaultTemplate = true,
+    onReady: () => void,
+    onError: () => void
+  ) {
+    this.resultTemplatesManager = buildResultTemplatesManager(
+      this.bindings.engine
+    );
+
+    const customTemplates = await Promise.all(
+      Array.from(elements).map(async (resultTemplateElement) => {
+        const template = await resultTemplateElement.getTemplate();
+        if (!template) {
+          onError();
+        }
+        return template;
+      })
+    );
+    const templates = (
+      includeDefaultTemplate ? [this.makeDefaultTemplate()] : []
+    ).concat(
+      customTemplates.filter(identity) as ResultTemplate<DocumentFragment>[]
+    );
+
+    this.resultTemplatesManager.registerTemplates(...templates);
+    onReady();
+  }
+
+  public getResultId(
+    result: Result | FoldedCollection,
+    resultListState: FoldedResultListState | ResultListState
+  ) {
+    return (
+      this.getUnfoldedResult(result).uniqueId + resultListState.searchResponseId
+    );
+  }
+
+  public getUnfoldedResult(result: Result | FoldedResult): Result {
+    return (result as FoldedResult).result ?? result;
+  }
+
+  public handleInfiniteScroll(
```

Comment:
I think it would be simpler to understand if this became a getter read by both result lists.
Example:
```ts
public get hasReachedEndOfList() {
    return window.innerHeight + window.scrollY >= host.offsetHeight;
}
```

Reply:
has this ever worked? I did some logging and noticed that it was always saying that it was true, because `host.offsetHeight === 0`. host only has a shadow dom, so I'm changing it to `this.host.shadowRoot?.firstElementChild.offsetHeight` which seems to work. lmk if there are any issues with it

Reply:
as far as I'm aware, this was supposed to work 🤔 
If this fixes it, them your change sounds good